mono/mini/mini-runtime.c

```diff
@@ -2597,8 +2597,24 @@ mono_llvmonly_fallback_imt_tramp (gpointer *arg, MonoMethod *imt_method)
 {
 	int i = 0;
 
-	while (arg [i] && arg [i] != imt_method)
-		i += 2;
+	while (arg [i]) {
```

Comment:
Haven’t verified this, but I think a `for` would be more readable here. Something like:

```
for (; arg [i]; i += 2) {
    MonoMethod *candidate = arg [i];
    if (candidate == imt_method)
        break;
    if (imt_method->klass->is_array_special_interface
        && candidate->klass->is_array_special_interface
        && mono_class_get_generic_class (imt_method->klass)->container_class
            == mono_class_get_generic_class (candidate->klass)->container_class
        && ((MonoMethodInflated *)imt_method)->declaring
            == ((MonoMethodInflated *)candidate)->declaring)
        break;
}
```